src/Onboarding/Routes/CurrencyRoute.php

```diff
@@ -80,36 +80,22 @@ public function registerRoute()
                         'value' => [
                             'type' => 'string',
                             'required' => true,
-                            // 'validate_callback' => [ $this, 'validateSetting' ],
+                            'validate_callback' => [$this, 'validateSetting'],
                             'sanitize_callback' => 'sanitize_text_field',
                         ],
                     ],
                 ],
-                'schema' => [$this, 'getSchema'],
             ]
         );
     }
 
     /**
-     * @since 2.8.0
-     * @return array
+     * Limits the symbol to a 2-letter country code
      *
+     * @unreleased
      */
-    public function getSchema()
+    public function validateSetting($value): bool
     {
-        return [
-            // This tells the spec of JSON Schema we are using which is draft 4.
-            '$schema' => 'http://json-schema.org/draft-04/schema#',
-            // The title property marks the identity of the resource.
-            'title' => 'onboarding',
-            'type' => 'object',
-            // In JSON Schema you can specify object properties in the properties attribute.
-            'properties' => [
-                'currencyCode' => [
-                    'description' => esc_html__('Two letter code representing a country.', 'give'),
-                    'type' => 'string',
-                ],
-            ],
-        ];
+        return preg_match('/^[A-Z]{2}$/i', $value) === 1;
```

Comment:
@JasonTheAdams Can you explain why we are validating the country code for the currency rest api endpoint?

Reply:
I am getting 400 HTTP errors when setup the donation form during onboarding.

<img width="1084" alt="image" src="https://user-images.githubusercontent.com/1784821/177805626-de66adb8-8ea2-4678-8c2c-1097de01ecbd.png">

Reply:
Good catch @ravinderk , I think the countryCode which uses 2 letter code was mixed up here. The currencyCode needs to be 3 letters